architecture/2025-04-aci-confidential-containers.md

```diff
@@ -0,0 +1,388 @@
+pivot in the future. This is a good place to cover risks.
+-->
+
+The CCE policy can only be generated using the `confcon` CLI extension. There are no available SDKs to generate this value. As a result, Radius must have access to the extension to generate policies on behalf of the user. 
+
+### API design (if applicable)
+
+<!--
+Include if applicable – any design that changes our public REST API, CLI
+arguments/commands, or Go APIs for shared components should provide this
+section. Write N/A here if not applicable.
+- Describe the REST APIs in detail for new resource types or updates to
+  existing resource types. E.g. API Path and Sample request and response.
+- Describe new commands in the CLI or changes to existing CLI commands.
+- Describe the new or modified Go APIs for any shared components.
+-->
+
+`typespec/Applications.Core/containers.tsp`
+
+```typespec
+@doc("The properties for runtime configuration")
+model RuntimesProperties {
+  @doc("The runtime configuration properties for Kubernetes")
+  kubernetes?: KubernetesRuntimeProperties;
+
+  @doc("The runtime configuration properties for ACI")
+  aci?: ACIRuntimeProperties;
+}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("A merge will be applied to the ContainerGroupProfile object when this container is being deployed.")
+model ContainerGroupProfile is Record<unknown>;
+
+@doc("The runtime configuration properties for ACI")
+model ACIRuntimeProperties {
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type"
+  @doc("A merge will be applied to the ContainerGroupProfile object when this container is being deployed.")
+  containergroupprofile?: ContainerGroupProfile;
+}
+```
+
+### CLI Design (if applicable)
+<!--
+Include if applicable – any design that changes Radius CLI
+arguments/commands. Write N/A here if not applicable.
+- Describe new commands in the CLI or changes to existing CLI commands.
+-->
+
+`rad init` would add 2 steps: one step to determine which compute platform to use when the user chooses to set up an environment and one step to choose confidential containers. See the user experience section for details. 
+
+### Error Handling
+<!--
+Describe the error scenarios that may occur and the corresponding recovery/error handling and user experience.
+-->
+
+Radius would expect the `runtimes` property to have the correct format. We would return errors if the formatting is incorrect (this would be surfaced during the merge process) or when deploying the container group profile (this would be surfaced by ARM if the ACI SDK throws an error on creation). 
+
+## Test plan
+
+<!--
+Include the test plan to validate the features including the areas that
+need functional tests.
+
+Describe any functionality that will create new testing challenges:
+- New dependencies
+- External assets that tests need to access
+- Features that do I/O or change OS state and are thus hard to unit test
+-->
+
+TBD
+
+## Security
```

Comment:
Do we require users to setup azure credentials on the client side with this change? Is it limited to platform engineers setting up the environment or also the dev deploying the app?

Reply:
We would still require Azure credentials since we're deploying Azure resources, but there shouldn't be an additional credential setup needed from this change